### DISCUSSION TITLE
Please add device shortcode ("PIN") lock support

### DISCUSSION THREAD

**Discussion by eugenesvk** [ente-io/ente#1037] [Enhancements]
As an alternative to Touch ID like in the Authy app
Doesn't need to be numerical (also, for the upcoming desktop app alpha app locks are more convenient to type)

**Comment by w21dk** [comment]
seconded. I don't want the device's passcode to be able to unlock the 2FA app. Very poor security practice. Much better to have a separate PIN for the 2FA app.

From a security perspective: if someone has your phone and the phone's passcode, they then also have your 2FA codes.

Ideally you should be able to unlock the 2FA app with biometric or a separate PIN (6+ digits/characters) that is not the passcode.

However if the biometric falls back to passcode in the event of multiple biometric failures, then this defeats the purpose of having a separate PIN.
In that scenario, I would disable biometric/passcode and only use a separate PIN to unlock the 2FA app.

**Comment by Larch35** [comment]
Agreed. I was really surprised to see this feature is missing. Would be great if this could be added as a priority... it's the main thing that is stopping me from switching from another authenticator app.

**Comment by daeh** [comment]
Adding my enthusiastic support to be able to set an ente-auth specific pin that's required to open the app (with maybe a variable lockout delay). Thanks for all the development effort!!
